### PR TITLE
Manipulation demo added

### PR DESCRIPTION
**Pull request by juandpenan**
Manipulation demo added as well as the implementation of several BT nodes:

- ExtractObjectsFromScene
- GenerateTextFromObjects
- PickObject

the instruction to run the demo were also added to the readme

### CODE REVIEW

File: bt_nodes/perception/CMakeLists.txt [hunk 1]
```diff
@@ -24,14 +39,34 @@ set(dependencies
...
+ament_target_dependencies(${bt_plugin} ${dependencies})
+target_compile_definitions(${bt_plugin} PRIVATE BT_PLUGIN_EXPORT)
 endforeach()
 
+install(TARGETS
+${plugin_libs}
+EXPORT ${PROJECT_NAME}
+ARCHIVE DESTINATION lib
+LIBRARY DESTINATION lib
+RUNTIME DESTINATION lib/${PROJECT_NAME}
```
Comment (Juancams) [inline bt_nodes/perception/CMakeLists.txt:58]: Duplicated

File: bt_nodes/perception/CMakeLists.txt [hunk 2]
```diff
@@ -7,15 +7,30 @@ endif()
...
 set(dependencies
   rclcpp
+  rclpy
+  behaviortree_cpp_v3
+  yolov8_msgs
+  moveit_msgs
+  shape_msgs
+  tf2_ros
+  geometry_msgs
+  tf2_geometry_msgs
```
Comment (Juancams) [inline bt_nodes/perception/CMakeLists.txt:32]: Duplicated

File: bt_nodes/perception/src/IsDetected.cpp
Comment (Juancams) [inline bt_nodes/perception/src/IsDetected.cpp:1]: Delete this pls

File: bt_nodes/bt_test/CMakeLists.txt [hunk 3]
```diff
@@ -31,6 +34,9 @@ set(dependencies
   perception
   tf2_ros
   hri
+  backward_ros
```
[the remark this answers is not in the document]
Comment (Juancams) [reply bt_nodes/bt_test/CMakeLists.txt:37]: Ok no, the error is that you have to add its find_package

File: bt_nodes/perception/src/IsDetected.cpp
Comment (juandpenan) [reply bt_nodes/perception/src/IsDetected.cpp:1]: Deleted

File: bt_nodes/perception/CMakeLists.txt (hunk 2 above)
Comment (juandpenan) [reply bt_nodes/perception/CMakeLists.txt]: Duplicate msgs deleted

File: bt_nodes/bt_test/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (juandpenan) [reply bt_nodes/bt_test/CMakeLists.txt]: backward ros removed

**Pull request merged**